### PR TITLE
[WIP] Proximity fixed point values

### PR DESCRIPTION
**Pull request by dnlup**
Add proximity devices resolutions.

### CODE REVIEW

File: lib/proximity.js [hunk 1]
```diff
@@ -163,7 +176,15 @@ var Controllers = {
...
-        return raw;
+        //
+        // Datasheet available at https://www.maxbotix.com/documents/XL-MaxSonar-EZ_Datasheet.pdf
+        //
+        // From intro on page 1
+        // 'The MB1200 and MB1300 sensor series detects objects from 0-cm1
+        //  to 765-cm (25.1 feet) or 1068cm (35 feet) (select models) and
+        // provide sonar range information from 20-cm2
+        //  out to765-cm or 1068-cm (select models) with 1-cm resolution...'
+        return toFixed(raw, 0);
```
Comment (dnlup) [inline lib/proximity.js]: @rwaldron  Here I forced the call to `toFixed` even if not needed. Maybe it is too much. I can revert it if you prefer.

File: lib/proximity.js [hunk 2]
```diff
@@ -276,7 +304,12 @@ var Controllers = {
     },
     toCm: {
       value: function(raw) {
-        return raw;
+        // Datasheet available at http://www.robotshop.com/media/files/pdf/operating-manual-llm20c132i500s011.pdf
+        //
+        // From `Technology` paragraph at page 11
+        // 'Our patented, high"accuracy"
+        // measurement"technique"enables"distance"measurement"accuracy down"to 1cm..'
+        return toFixed(raw, 0);
```
Comment (dnlup) [inline lib/proximity.js]: @rwaldron Same as https://github.com/rwaldron/johnny-five/pull/1439/files#diff-c5fe53df19f93fb5988d5336a99e42a9R187.

**Comment by dnlup** [comment]
Sorry, but I could not find valuable data on `EV3` and `SRF10` devices, plus many analog ones:
* GP2Y0A21YK
* GP2D120XJ00F
* GP2Y0A02YK0F
* GP2Y0A41SK0F
* GP2Y0A710K0F

File: lib/proximity.js (hunk 1 above)
Comment (rwaldron) [reply lib/proximity.js]: How about an optimization? Truncate the decimal digits by shifting the bits off: `raw >> 0`, this shifts right by `0` bits, but first it must convert the left hand operand to an int32—which means discarding the exponent bits :)

**Comment by rwaldron** [comment]
This is excellent work! I've checked that travis failure and it appears to be unrelated to these changes (also, it passes locally)

**Comment by rwaldron** [comment]
Landed!

**Comment by dnlup** [comment]
@rwaldron Thanks!
Travis had problems with osX builds at that time, it happened that more than more PR failed because of it :smile: .

File: lib/proximity.js (hunk 1 above)
Comment (dnlup) [reply lib/proximity.js]: I'll do it and re open this PR, thanks!

**Pull request closed without merging**